docs/migration-guides.md

```diff
@@ -0,0 +1,1306 @@
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
+
+...
+
+arc4.abiCall<typeof MyContract.prototype.myMethod>({ ... })
+```
+
+This form of invocation supports using type only imports `import type { MyContract } from '...'` which allow for circular references versus value imports which cannot be circular.
+
+If, in some unforeseen circumstances, both a type argument and the `method` property are provided to the `abiCall` helper, the type argument takes precedence and the method specified by the type argument is called.
+
+**Before (Algorand TypeScript beta)**
+
+```ts
+import { arc4, assert } from '@algorandfoundation/algorand-typescript'
+import Hello, { HelloStubbed } from './Hello.algo'
+
+...
+
+const result2 = arc4.abiCall(Hello.prototype.greet, {
+  appId: 1234,
+  args: ['abi'],
+}).returnValue;
+
+assert(result2 === 'hello abi');
+
+const result3 = arc4.abiCall(HelloStubbed.prototype.greet, {
+  appId: 1234,
+  args: ['stubbed'],
+}).returnValue;
+assert(result3 === 'hello stubbed');
+```
+
+**After (Algorand TypeScript 1.0)**
+
+```ts
+import { arc4, assert } from '@algorandfoundation/algorand-typescript'
+import Hello, { HelloStubbed } from './Hello.algo'
+
+...
+
+// provide method property
+const result2 = arc4.abiCall({
+  method: Hello.prototype.greet,
+  appId: app,
+  args: ['abi'],
+}).returnValue;
+
+assert(result2 === 'hello abi');
+
+// or provide type argument, this approach supports using type only imports
+const result3 = arc4.abiCall<typeof HelloStubbed.prototype.greet>({
+  appId: app,
+  args: ['stubbed'],
+}).returnValue;
+assert(result3 === 'hello stubbed');
+```
+
+#### Rename `interpretAsArc4` function to `convertBytes`
+
+The function now accepts an `options` object with `strategy: 'unsafe-cast' | 'validate'` and an optional `prefix` parameter.
+
+**Before (Algorand TypeScript beta)**
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+
+...
+
+const x = arc4.interpretAsArc4<Uint<32>>(a);
+const y = arc4.interpretAsArc4<Byte>(b, 'log');
+```
+
+**After (Algorand TypeScript 1.0)**
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+
+...
+
+const x = arc4.convertBytes<arc4.Uint<32>>(a, { strategy: 'validate' });
+const y = arc4.convertBytes<arc4.Byte>(b, { prefix: 'log', strategy: 'unsafe-cast' });
+```
+
+#### Remove `BoxRef`, use `Box<bytes>` instead
+
+`Box<bytes>` now includes all functionality previously available in `BoxRef`, including `extract`, `replace`, `resize`, and `splice` methods.
+
+**Before (Algorand TypeScript beta)**
+
+```ts
+import { Contract, BoxRef, Bytes, bytes } from '@algorandfoundation/algorand-typescript'
+
+...
+
+const box = BoxRef({ key: 'test_key' });
+box.create({ size: 32768 });
+
+const boxValue = new Uint8Array(Array(5).fill([0x01, 0x02]).flat());
```

Comment:
@joe-p is using the uint8array constructor in a contract valid? I assume not, lmk if this needs to be changed.